superset/assets/javascripts/explore/components/ChartContainer.jsx

```diff
@@ -248,71 +249,72 @@ class ChartContainer extends React.PureComponent {
-              }
-
-              <div className="pull-right">
-                {this.props.chartStatus === 'success' &&
-                this.props.queryResponse &&
-                this.props.queryResponse.is_cached &&
-                  <CachedLabel
-                    onClick={this.runQuery.bind(this)}
-                    cachedTimestamp={queryResponse.cached_dttm}
-                  />
+                      <a
+                        className="edit-desc-icon"
+                        href={`/slicemodelview/edit/${this.props.slice.slice_id}`}
+                      >
+                        <i className="fa fa-edit" />
+                      </a>
+                    </TooltipWrapper>
+                  </span>
                 }
-                <Timer
-                  startTime={this.props.chartUpdateStartTime}
-                  endTime={this.props.chartUpdateEndTime}
-                  isRunning={this.props.chartStatus === 'loading'}
-                  status={CHART_STATUS_MAP[this.props.chartStatus]}
-                  style={{ fontSize: '10px', marginRight: '5px' }}
-                />
-                <ExploreActionButtons
-                  slice={this.state.mockSlice}
-                  canDownload={this.props.can_download}
-                  chartStatus={this.props.chartStatus}
-                  queryResponse={queryResponse}
-                  queryEndpoint={getExploreUrl(this.props.latestQueryFormData, 'query')}
-                />
+                <div className="pull-right">
+                  {this.props.chartStatus === 'success' &&
+                  this.props.queryResponse &&
+                  this.props.queryResponse.is_cached &&
+                    <CachedLabel
+                      onClick={this.runQuery.bind(this)}
+                      cachedTimestamp={queryResponse.cached_dttm}
+                    />
+                  }
+                  <Timer
+                    startTime={this.props.chartUpdateStartTime}
+                    endTime={this.props.chartUpdateEndTime}
+                    isRunning={this.props.chartStatus === 'loading'}
+                    status={CHART_STATUS_MAP[this.props.chartStatus]}
+                    style={{ fontSize: '10px', marginRight: '5px' }}
+                  />
+                  <ExploreActionButtons
+                    slice={this.state.mockSlice}
+                    canDownload={this.props.can_download}
+                    chartStatus={this.props.chartStatus}
+                    queryResponse={queryResponse}
+                    queryEndpoint={getExploreUrl(this.props.latestQueryFormData, 'query')}
+                  />
+                </div>
               </div>
-            </div>
-          }
-        >
-          {this.renderChart()}
-        </Panel>
+            }
+          >
+            {this.renderChart()}
+          </Panel>
+        </div>
+        {this.props.viz_type === 'line_ttest' &&
+        <PairedTTestTableContainer />}
```

Comment:
New visualizations should not live here. You probably want to call `ReactDom.render` from within `nvd3.jsx`. No need to touch this file at all.